docker/package/wizard_structure.py

```diff
@@ -715,13 +715,16 @@ def import_key(self, key_mode_query, ledger_app=None):
                         f"Waiting for funds to arrive... (Ctrl + C to choose another option)."
                     )
                     try:
-                        # dry-run delegate registration until it succeeds
                         while True:
                             result = get_proc_output(
                                 f"sudo -u tezos {suppress_warning_text} octez-client {tezos_client_options} "
-                                f"register key {baker_alias} as delegate --dry-run"
+                                f"get balance for {baker_alias}"
                             )
-                            if result.returncode == 0:
+                            # workaround until https://gitlab.com/tezos/tezos/-/issues/4200 resolved
+                            if result.returncode == 0 and (
+                                int(re.search(b"[0-9]+", result.stdout).group(0))
+                                >= 6000
```

Comment:
There's a few problems with this, because AFAIK 6000 is needed for baking (and baking only), but there's more to keep in mind.

If the delegate has 6000 tez but hasn't been registered yet then the next operation in the setup wizard (the delegate registration itself) will put the balance below the minimum (due to fees) needed for baking, which is not great.

OTOH if the delegate has been registered already, 6000 is perfectly fine for baking, so this logic is "correct".

However it really isn't the best way to do this either way, because one can _have_ less than 6000 tez, but still be able to bake. For example one might have a very low balance (IIRC 600 might be the minimum) but others have delegated to this address. This still works and isn't an impediment to baking... :confused: 
So IMO the logic based on balance is not the best for this workaround.

Another point is that AFAIK (but please check, I didn't make sure) voting doesn't require 6000 tez and this function here is also called in the voting wizard. 

As I mentioned elsewhere maybe we should drop the `--dry-run` and stop as soon as the registration command returns successfully (either because it registered the baker now or because the baker was registered already).